test/OpenTelemetry.Exporter.Geneva.Tests/LogSerializationTests.cs

```diff
@@ -0,0 +1,149 @@
+// <copyright file="LogSerializationTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Logs;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace OpenTelemetry.Exporter.Geneva.Tests;
+
+public class LogSerializationTests
+{
+    /*
+    Run from the current directory:
+    dotnet test -f net6.0 --filter FullyQualifiedName~LogSerializationTests -l "console;verbosity=detailed"
+    */
+    private readonly ITestOutputHelper output;
+
+    public LogSerializationTests(ITestOutputHelper output)
+    {
+        this.output = output;
+    }
+
+    [Fact]
+    public void SerializationTestForException()
+    {
+        var exceptionMessage = "Exception Message";
+        var exportedFields = GetExportedFieldsAfterLogging(logger => logger.Log<object>(
+            logLevel: LogLevel.Information,
+            eventId: default,
+            state: null,
+            exception: new Exception(exceptionMessage),
+            formatter: null));
+
+        PrintFields(this.output, exportedFields);
+        var actualExceptionMessage = exportedFields["env_ex_msg"];
+        Assert.Equal(exceptionMessage, actualExceptionMessage);
+    }
+
+    private static void PrintFields(ITestOutputHelper output, Dictionary<object, object> fields)
+    {
+        foreach (var field in fields)
+        {
+            output.WriteLine($"{field.Key}:{field.Value}");
+        }
+    }
+
+    private static Dictionary<object, object> GetExportedFieldsAfterLogging(Action<ILogger> doLog)
```

Comment:
This wouldn't help us cover a lot of the specific scenarios which need certain exporter or logger options to be set such as: `CustomFields`, `TableNameMappings`, `ParseStateValues`, `IncludeScopes` etc.

We could probably update this method's signature to take in a `bool` parameter for each of those options and create the exporter and logger accordingly.

Reply:
Yes. This can be modified to take Action<GenevaExporterOptions> to control the behavior.

Reply:
Do you like the approach in this PR ? If yes, I can merge and iterate on it. (And move other tests to this pattern and cleanup)